win32/tests/gballoc_ll_jemalloc_ut/gballoc_ll_jemalloc_ut.c

```diff
@@ -628,4 +633,100 @@ TEST_FUNCTION(gballoc_ll_print_stats_calls_the_print_function_with_NULL_does_not
     ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
 }
 
+/* gballoc_ll_set_decay */
+
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_001: [ gballoc_ll_set_decay shall advance jemalloc's internal epoch counter. ]*/
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_002: [ gballoc_ll_set_decay shall set the dirty decay time for new arenas to decay_milliseconds milliseconds. ]*/
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_003: [ gballoc_ll_set_decay shall set the muzzy decay time for new arenas to decay_milliseconds milliseconds. ]*/
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_004: [ gballoc_ll_set_decay shall fetch the number of existing jemalloc arenas. ]*/
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_005: [ For each existing arena: ]*/
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_006: [ gballoc_ll_set_decay shall set the dirty decay time for the arena to decay_milliseconds milliseconds. ]*/
+/* Tests_SRS_GBALLOC_LL_JEMALLOC_28_007: [ gballoc_ll_set_decay shall set the muzzy decay time for the arena to decay_milliseconds milliseconds. ]*/
+TEST_FUNCTION(gballoc_ll_set_decay_succeeds)
+{
+    ///arrange
+    uint64_t num_arenas = NUM_ARENAS;
+
+    char* epoch;
+    char* dirty_decay_ms;
+    char* muzzy_decay_ms;
+    char* narenas;
+
+    STRICT_EXPECTED_CALL(mock_je_mallctl(IGNORED_ARG, NULL, NULL, NULL, 0))
+        .CaptureArgumentValue_name(&epoch)
+        .SetReturn(0);
+    STRICT_EXPECTED_CALL(mock_je_mallctl(IGNORED_ARG, NULL, NULL, IGNORED_ARG, IGNORED_ARG))
+        .CaptureArgumentValue_name(&dirty_decay_ms)
+        .SetReturn(0);
+    STRICT_EXPECTED_CALL(mock_je_mallctl(IGNORED_ARG, NULL, NULL, IGNORED_ARG, IGNORED_ARG))
+        .CaptureArgumentValue_name(&muzzy_decay_ms)
+        .SetReturn(0);
+    STRICT_EXPECTED_CALL(mock_je_mallctl(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG))
+        .CaptureArgumentValue_name(&narenas)
+        .CopyOutArgumentBuffer_oldp(&num_arenas, sizeof(num_arenas))
+        .SetReturn(0);
+    for (uint64_t i = 0; i < NUM_ARENAS; i++)
+    {
+        STRICT_EXPECTED_CALL(mock_je_mallctl(IGNORED_ARG, NULL, NULL, IGNORED_ARG, IGNORED_ARG))
+            .SetReturn(0);
+        STRICT_EXPECTED_CALL(mock_je_mallctl(IGNORED_ARG, NULL, NULL, IGNORED_ARG, IGNORED_ARG))
+            .SetReturn(0);
+    }
+
+    ///act
+    int result = gballoc_ll_set_decay(42);
```

Comment:
>42 [](http://example.com/codeflow?start=38&length=2)

this "42" here should be asserted that it is passed "somewhere" in the mallctl function calls.

Reply:
Yeah we are doing it using ValidateArgumentBuffer